### PR TITLE
Clean up ci-runner dockerfiles & upgrade to nodejs 14.19.1

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
 Clean up ci-runner dockerfiles & upgrade to nodejs 14.19.1.
* Remove AL2 runner, as it is not used and deprecated.
* Change 14.18.2 to 14.19.1 nodejs version per this: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1487
* Removed nodejs and cypress from the opensearch build runner as it is not needed anymore.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
New images:

(Will show up once the build is complete)

* https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&ordering=last_updated&name=ci-runner-centos7-opensearch-build-v2
* https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&ordering=last_updated&name=ci-runner-centos7-opensearch-dashboards-build-v2
* https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&ordering=last_updated&name=ci-runner-rockylinux8-opensearch-dashboards-integtest-v2

All images now use the number v2.

File: jenkins/docker/Jenkinsfile [hunk 1]
```diff
@@ -40,7 +40,6 @@ pipeline {
             }
             steps {
                 script {
-                    currentBuild.description = DOCKER_BUILD_SCRIPT_WITH_COMMANDS
```
Comment (tianleh) [inline jenkins/docker/Jenkinsfile:43]: Is this change intentional?

Comment (peterzhuamazon) [reply jenkins/docker/Jenkinsfile:43]: yes, too long.

File: docker/ci/dockerfiles/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile [hunk 2]
```diff
@@ -66,7 +66,7 @@ WORKDIR /usr/share/opensearch
 # Hard code node version and yarn version for now
 # nvm environment variables
 ENV NVM_DIR /usr/share/opensearch/.nvm
```
Comment (tianleh) [inline docker/ci/dockerfiles/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile:68]: can we add `jq` support https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/152#issuecomment-1080951494

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile:68]: Added now.

**Comment by tianleh** [comment]
I can perform some sanity check on the new images when they are ready. @peterzhuamazon

**Comment by peterzhuamazon** [comment]
Everything is built right now all the images above updated to v2.
Cypress on arm64 can now automatically compile during docker image build time.
Thanks.

**Review by tianleh — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Image too big change to multi-stage now.
```
opensearchstaging/ci-runner               ci-runner-rockylinux8-opensearch-dashboards-integtest-v2   daf9ba3a8c03   43 minutes ago   14.5GB

```

**Review by tianleh — DISMISSED** [review]
Please optimize the image sizes before merging.

**Comment by peterzhuamazon** [comment]
Size fixed now:
```
[ci-runner-rockylinux8-opensearch-dashboards-integtest-v2](https://hub.docker.com/layers/ci-runner/opensearchstaging/ci-runner/ci-runner-rockylinux8-opensearch-dashboards-integtest-v2/images/sha256-1054f5782bbf282d9373dba556c943fb7aa9dd2b559d0d69efaee0a30c89b5f3?context=explore)
Last pushed 7 minutes ago by [opensearchstaging](https://hub.docker.com/u/opensearchstaging)
Digest
OS/ARCH
Compressed Size
1054f5782bbf
linux/amd64
1.05 GB
4d3830db21d0
linux/arm64
1.04 GB
```

**Comment by peterzhuamazon** [comment]
```
[opensearch@c28e1e31fe3e ~]$ cypress run
It looks like this is your first time using Cypress: 9.5.4

✔  Verified Cypress! /usr/share/opensearch/.cache/Cypress/9.5.4/Cypress

Opening Cypress...
[1773:0423/103729.554624:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is
Could not find a Cypress configuration file in this folder: /usr/share/opensearch
[opensearch@c28e1e31fe3e ~]$ uname -a
Linux c28e1e31fe3e 4.14.273-207.502.amzn2.aarch64 #1 SMP Wed Mar 30 01:13:12 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

Ignore failure as there is no cypress test present.
This proves that on aarch64 the compiled cypress runs correctly.

Thanks.

**Comment by peterzhuamazon** [comment]
Rebuilt with cmake installed so no module compilation failure, still the sandbox test seems error out due to we are building a docker image inside a docker, but binary built.

**Comment by peterzhuamazon** [comment]
@tianleh please review.
Thanks.

**Comment by prudhvigodithi** [comment]
Hey @peterzhuamazon I see `ci-runner` is mentioned twice in the image `opensearchstaging/ci-runner:ci-runner-centos7-opensearch-build-v2`, can we have the image like `opensearchstaging/ci-runner-centos7:2.0`, would this make simpler ?
Thank you

**Comment by peterzhuamazon** [comment]
> Hey @peterzhuamazon I see `ci-runner` is mentioned twice in the image `opensearchstaging/ci-runner:ci-runner-centos7-opensearch-build-v2`, can we have the image like `opensearchstaging/centos7-ci-runner:2.0`, would this make simpler ? Thank you

It is always like that and I understand where you come with that question.
It is not ideal but we have been putting all the ci runner images in the same repo without creating multiple other repos.
In the next iteration we can try something else, at least standardize the image names, which is still not there yet.

Let's not do too much change now as release is next week and a lot of coding is depending on these names (yes there are code dependencies on these names).

Plus even with the help of Jenkins these images build in 3 hours due to multiple compilation on the fly through binary translation (which is by design to do so). So I would not start that now.

There is an issue in this repo already pointing that out.

Thanks.

**Review by tianleh — APPROVED** [review]

**Comment by gaiksaya** [comment]
Hey @peterzhuamazon if you have issue for enhancement for ci images can you add this example as well? https://github.com/opensearch-project/opensearch-ci/pull/107
so that we don't have to wait for someone to build the image, whenever a change is pushed to this repo, the ci image can be built automatically. 
Thanks!

**Comment by peterzhuamazon** [comment]
> Hey @peterzhuamazon if you have issue for enhancement for ci images can you add this example as well? [opensearch-project/opensearch-ci#107](https://github.com/opensearch-project/opensearch-ci/pull/107) so that we don't have to wait for someone to build the image, whenever a change is pushed to this repo, the ci image can be built automatically. Thanks!

Not necessarily the same as we have other script to setup env before dockerfile being built, thus cannot directly run docker build from github actions. Please the compiling is intense and it would crash if we do it on the 2 core machine of github runners. It would also take more than 6 hours even if it runs.
